### ISSUE TITLE
Wrong syntax highlighting for function signature

### ISSUE THREAD

**Issue by jfcherng** [sublimelsp/LSP-pyright#233]
![image](https://user-images.githubusercontent.com/6594915/225525929-c06e006c-ffd0-40a2-b1ee-1d49e6cac1e5.png)

```py
(function) def foo(callback: ((_T@foo) -> bool) | None = None) -> (_T@foo | None)
#                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Also, I would like to `foo`s in `_T@foo` have a special scope.

**Comment by rchl** [comment]
I'm assuming that something like `((_T@foo) -> bool)` is not a valid syntax in this context (or anywhere)? Which makes me wonder why try to handle a syntax that is clearly not valid Python syntax. It is bound to fail since there are no rules and regulations on what the server can use.

But if you have the strength and will to fight against this then feel free.

**Comment by jfcherng** [comment]
> I'm assuming that something like `((_T@foo) -> bool)` is not a valid syntax in this context (or even anywhere)? Which makes me wonder why do we try to handle a syntax that is clearly not valid Python syntax. It is bound to fail since there are no rules and regulations on what the server can use.

Is this the point that we have this file? https://github.com/sublimelsp/LSP-pyright/blob/acf63fe3f1879521eef4a99dd01ad95c456bdf57/syntaxes/pyright.sublime-syntax

Otherwise, we just use exact Python syntax for the signature popup and for any weird highlighting we just say it's not Python.

**Comment by rchl** [comment]
yeah but initially my understanding was that it is only the prefix that we have to get rid of and the rest will be a valid python syntax which would be trivial to handle. Now it's turning into a completely different language :)
